src/Adapter/Shop/Url/ProductPreview.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Shop\Url;
+
+use Context;
+use Link;
+use Tools;
+
+class ProductPreview
+{
+    /**
+     * @var Link
+     */
+    private $link;
+
+    public function __construct(Link $link)
+    {
+        $this->link = $link;
+    }
+
+    public function getPreviewUrl(int $productId, bool $active): string
+    {
+        $preview_url = $this->link->getProductLink($productId);
+        if (!$active) {
+            $context = Context::getContext();
+            $token = Tools::getAdminTokenLite('AdminProducts');
+
+            $admin_dir = dirname($_SERVER['PHP_SELF']);
```

Comment:
There might be a more reliable way to find the admin directory 😅

Reply:
amin directory seems to be not used anymore. I remove it and the feature works as usual ;-)